src/main/java/practice/StreamPractice.java

```diff
@@ -59,7 +91,14 @@ public Double getOddNumsAverage(List<Integer> numbers) {
      * return the names of all cats whose owners are women from `femaleAge` years old inclusively.
      */
     public List<String> getCatsNames(List<People> peopleList, int femaleAge) {
-        return Collections.emptyList();
+        Predicate<People> catsOwners =
+                people -> people.getSex() == People.Sex.WOMEN && people.getAge() >= femaleAge;
+        return peopleList.stream()
+                .filter(catsOwners)
+                .map(People::getCats)
+                .flatMap(Collection::stream)
```

Comment:
Optional but looks better
```suggestion
                .flatMap(people -> people.getCats().stream())
```